app/controllers/api/v2/hosts_controller.rb

```diff
@@ -112,6 +112,7 @@ def create
         @host = Host.new(host_attributes(host_params))
         @host.managed = true if (params[:host] && params[:host][:managed].nil?)
         apply_compute_profile(@host)
+        @host.suggest_default_pxe_loader if params[:host][:pxe_loader].nil?
```

Comment:
The UI works fine but V2 API (both host and hostgroup) do not.

On this line, it looks like:

```
@host.try(:operatingsystem)
=> nil
@host.hostgroup.operatingsystem
=> #<CentOS7...>
```

This is not likely related to your change, but can you fix this please? We need to take hostgroup OS into account. Also a test for this would help to prevent this. We have a couple, but not for this hostgroup use case.